index.js

```diff
@@ -129,7 +208,19 @@ function filterByGenre() {}
       }
     ];
  */
-function getAllMoviesReleasedAtOrBeforeYear() {}
+function getAllMoviesReleasedAtOrBeforeYear(movies, year) {
+
+  movTitle = [];
+
+  for(let i = 0; i < movies.length; i++){
+    let date = movies[i].released.split(' ');
+    if(date[date.length-1] <= year){
+      movTitle.push(movies[i]);
+    }
```

Comment:
This is a bit of a tricky situation here as 216 is setting a value to a variable with a bit of complicated logic and then 217 is indexing the last item of that variable for a comparison.  
While its still clunky I would reccomend setting your varaible to the value you want to compare if we do not plan on using any other indexes of the `date` array.
` let date = movies[i].released.split(' ')[movies[i].released.split("").length-1] `
I'm not in love with either to be honest but I feel that putting all the logic in one line is a better solution if we _don't_ plant to ever use `date` as an array.

Reply:
thank you for the feedback, definitely going to give it another go!